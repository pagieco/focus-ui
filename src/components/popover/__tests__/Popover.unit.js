import { shallowMount } from '@vue/test-utils';
import Popover from '../components/Popover.vue';

const shallowWrap = (propsData = {}) => shallowMount(Popover, { propsData });

describe('Popover', () => {
  it('is a valid vue component', () => {
    expect(Popover).toBeAComponent();
  });

  it('matches the snapshot', () => {
    expect(shallowWrap().element).toMatchSnapshot();
  });

  it('uses the isOpen prop', () => {
    const wrapper = shallowWrap({ isOpen: true });

    expect(wrapper.vm.$data.isShown).toBeTruthy();
  });

  it('uses the placement prop', () => {
    // TODO
  });

  it('uses the offset prop', () => {
    // TODO
  });

  it('emits the open event', () => {
    const wrapper = shallowWrap({ isOpen: false });

    wrapper.vm.open();

    expect(wrapper.emitted('open')).not.toBeUndefined();
  });

  it('emits the close event', () => {
    const wrapper = shallowWrap({ isOpen: true });

    wrapper.vm.close();

    expect(wrapper.emitted('close')).not.toBeUndefined();
  });
});
