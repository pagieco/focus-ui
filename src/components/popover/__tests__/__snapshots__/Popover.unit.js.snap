// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover matches the snapshot 1`] = `
<div
  class="popover"
  role="dialog"
>
  <span />
</div>
`;
